src/components/symptom-shark/presentational/BulletChart/BulletChart.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useContext } from "react";
+import "./BulletChart.css"
+import { getDatesForMonth } from "../../../../helpers/date-helpers";
+import { DateRangeContext, SymptomConfiguration, TextBlock, TreatmentConfiguration, getDayKey, language } from "../../../..";
+import { SymptomSharkVisualizationContext } from "../../container/VisualizationCoordinator/VisualizationCoordinator";
+import { startOfMonth } from "date-fns";
+
+export interface SymptomSharkBulletChartProps {
+    intervalStart?: Date;
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export default function (props: SymptomSharkBulletChartProps) {
+    let visualizationContext = useContext(SymptomSharkVisualizationContext);
+    if (!visualizationContext) {
+        return <TextBlock innerRef={props.innerRef}>Error: Bullet Chart must be used inside a Symptom Shark Visualization Coordinator.</TextBlock>
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = props.intervalStart || startOfMonth(new Date());
+    if (dateRangeContext) {
+        intervalStart = dateRangeContext.intervalStart;
+    }
+
+    if (!visualizationContext.symptoms.length && !visualizationContext.treatments.length) {
+        return null;
+    }
+
+    var monthDays = getDatesForMonth(intervalStart.getFullYear(), intervalStart.getMonth());
+
+    var symptomSelected = function (date: Date, symptomId: string) {
+        var key = getDayKey(date);
+        if (!visualizationContext!.logEntries[key]) {
+            return false;
+        }
+        return visualizationContext!.logEntries[key].symptoms.find((s) => s.id == symptomId);
+    }
+
+    var treatmentSelected = function (date: Date, treatmentId: string) {
+        var key = getDayKey(date);
+        if (!visualizationContext!.logEntries[key]) {
+            return false;
+        }
+        return visualizationContext!.logEntries[key].treatments.find((s) => s.id == treatmentId);
+    }
+
+    var symptomTotal = function (symptom: SymptomConfiguration) {
+        var total = 0;
+        for (var i = 0; i < monthDays.length; i++) {
+            if (symptomSelected(monthDays[i], symptom.id)) {
+                total++;
+            }
+        }
+        return total;
+    }
+
+    var treatmentTotal = function (treatment: TreatmentConfiguration) {
+        var total = 0;
+        for (var i = 0; i < monthDays.length; i++) {
+            if (treatmentSelected(monthDays[i], treatment.id)) {
+                total++;
+            }
+        }
+        return total;
+    }
+
+    return <div ref={props.innerRef} className="ss-bullet-chart">
+        {visualizationContext.symptoms.length > 0 &&
+            <table cellPadding="0" cellSpacing="0">
+                <thead>
```

Comment:
The table is repeated below, and looks like it could be a standalone presentational component

Reply:
I think I agree with Renee here; I think this is a pretty cool visualization.  It seems like this could be made general in a pretty similar way to how we do our current presentational calendar with the day renderer callback?  Is the reason that you're not expecting this to be generic because it's intended output is in a generated PDF?  This is one of the few visualizations we have that pretty clearly doesn't have a good way to re-format to mobile, so therefore it's re-usability is more limited?

As a side note - when I Google the phrase "Bullet Chart", I get a bunch of results all related to https://en.wikipedia.org/wiki/Bullet_graph.  There's even a Google product called "Looker Studio" that uses "Bullet Chart" https://support.google.com/looker-studio/answer/7205385#zippy=%2Cin-this-article for this specific type of bar graph.  If I got no results back, I wouldn't be concerned, but it looks like this might be stepping on a relatively "industry standard" term?

Reply:
Oh yeah just because I haven't made something generic yet definitely doesn't mean I don't want to.  Totally agree this would be a cool generic component.  There's almost certainly a way to get it to look good on mobile too; like with the chats you can fit a month's worth of data so long as you skip rendering every other date.

I'll take a swing at it and see if it's unambiguous enough for a generic presentational component.  Otherwise I might punt for now to try to get this in sooner rather than later.

But yeah it does seem like i'm misusing the term.  It basically originally came from the idea of "bullet journals" like this: 
https://www.reddit.com/r/bulletjournal/comments/104mmw9/it_was_then_that_i_realized_maybe_i_shouldve_made/

Even as I look at "Bullet journals) it seems like that's a much broader term...I bet there is a more industry standard term for this kind of viz

Reply:
I re-worked this; there's now a generic "MonthlyEventMatrix" presentational component and this symptom shark component was just renamed to "SymptomMatrix"